cmd/gateway-main.go

```diff
@@ -117,6 +118,25 @@ func newGatewayConfig(accessKey, secretKey, region string) error {
 	return nil
 }
 
+func getGatewayEndpoint(arg string) (endPoint string, secure bool, err error) {
+	u, err := url.Parse(arg)
+	if err != nil {
+		return "", false, err
+	}
+
+	switch u.Scheme {
+	case "http":
+		return u.Host, true, nil
+	case "https":
+		return u.Host, false, nil
+	case "":
+		// url.Parse("play.minio.io") puts play.minio.io in u.Path
```

Comment:
In go1.8 this wouldn't happen anymore and an error will be generated, if we need to support a scenario like this. We should handle it like `net.SplitHostPort()` style.